### PR TITLE
login button works while keyboard is mounted

### LINKED ISSUE #160: login button click functionality not working while keyboard is mounted.
Issue: After entering input in the fields on the login page, when clicking the login button while the keyboard is still on the screen, the button functionality does not trigger as expected. Instead, the keyboard is dismissed. After the keyboard is dismissed, clicking the login button again works as expected.

Expected Solution: The login functionality of the login button should work with a single click, even when the keyboard is still on the screen.

https://github.com/user-attachments/assets/9ca51d6a-5448-4968-9d3a-7753ef124f41

### PR DESCRIPTION
**Pull request by AmarHadzic**
Fixes #160 

**What was changed**
The login button behavior was modified so that user's can login while the keyboard is still visible. Specifically, the `keyboardShoudPersistTaps` property was added to make sure that the login button responds to taps even while the keyboard is mounted.

**Why was it changed**
Previously, users were forced to tap the login button more than once when the keyboard was mounted, leading to a poor user experience. In the current behavior, the first tap would dismiss the keyboard, requiring a second tap to trigger the login. By making the login button functional on the first tap, users experience a more efficient and better login experience.

**How was it changed**
The `keyboardShoudPersistTaps` property was set to "handled" on the KeyboardAwareScrollView on line 123 on login screen.tsx, which ensures that taps are recognized and handled appropriately when the keyboard is mounted. This fix prevents the keyboard from being dismissed first and then the user having to tap the login button again to login. Instead, this requires only one tap.

### CODE REVIEW

File: lib/screens/loginScreens/LoginScreen.tsx [hunk 1]
```diff
@@ -99,90 +110,79 @@ const LoginScreen: React.FC<LoginProps> = ({ navigation, route }) => {
...
 
   return (
-    <Auth0Provider domain={"{yourDomain}"} clientId={"{yourClientId}"}>
-      <KeyboardAwareScrollView
-        contentContainerStyle={styles.container}
-        style={{ backgroundColor: "#F4DFCD" }}
+    <KeyboardAwareScrollView
+      contentContainerStyle={styles.container}
+      style={{ backgroundColor: "#F4DFCD" }}
+      keyboardShouldPersistTaps="handled"
```
Comment (ademDurakovic) [inline lib/screens/loginScreens/LoginScreen.tsx:123]: Is this the line that fixed it?

File: __tests__/LoginScreen.test.tsx [hunk 2]
```diff
@@ -12,4 +12,21 @@ describe("LoginScreen", () => {
...
+  it("triggers login function when login button is pressed while keyboard is mounted", () => {
+    // Mock the login function prop
+    const mockLoginFunction = jest.fn();
+
+    // Pass the mockLoginFunction as the login prop to the LoginScreen component
+    const wrapper = shallow(<LoginScreen login={mockLoginFunction} />);
+
+    // Simulate user input in username and password fields
+    wrapper.find('TextInput[placeholder="Username"]').simulate('changeText', 'testuser');
+    wrapper.find('TextInput[placeholder="Password"]').simulate('changeText', 'password123');
```
Comment (ademDurakovic) [inline __tests__/LoginScreen.test.tsx:25]: Good job adding tests

**Review by ademDurakovic — COMMENTED** [review]
Good simple solution

**Review by rcAsironman — COMMENTED** [review]
Overall, the implementation for Issue #160 is solid and adds valuable functionality to the project. Great work on addressing the core requirements effectively.

Please consider the suggestion to improve the user experience by dismissing the keyboard when the user hits the login button. Once this adjustment is made, the PR will be ready for merging.

Thank you for your hard work on this! If you have any questions or need further clarification, feel free to ask.

File: lib/models/user_class.ts [hunk 3]
```diff
@@ -54,40 +61,62 @@ export class User {
...
+      //   this.userData = data;
+      //   if (this.userData !== null) {
+      //     await this.persistUser(this.userData);
+      //   }
+      //   this.notifyLoginState();
+      //   console.log("From userClass, Data ***************************==>>************************************ ", this.userData)
+      //   return "success";
+      // } else {
+      //   throw new Error("There was a server error logging in.");
+      // }
```
Comment (kate-holdener) [inline lib/models/user_class.ts:95]: Please remove commented out code in the final version.

File: lib/navigation/AppNavigator.tsx [hunk 4]
```diff
@@ -44,29 +44,30 @@ const HomeStack = () => {
...
-        {(props: EditNoteProps) => <EditNote {...props} />}
-      </Stack.Screen>
+      />
     </Stack.Navigator>
   );
 };
 
 const AppNavigator: React.FC = () => {
-  const [navState, setNavState] = useState<"loading" | "onboarding" | "login" | "home">("loading");
+  // const [navState, setNavState] = useState<"loading" | "onboarding" | "login" | "home">("loading");
```
Comment (kate-holdener) [inline lib/navigation/AppNavigator.tsx:55]: Please don't leave commented out code

File: lib/screens/AddNoteScreen.tsx [hunk 5]
```diff
@@ -179,8 +179,8 @@ const AddNoteScreen: React.FC<AddNoteScreenProps> = ({ navigation, route }) => {
           document.getElementById('videoElement').addEventListener('play', function(e) {
             // Preventing the rich text editor from gaining focus when the video is played
             e.preventDefault();
-            // Assuming you have a way to send a message to your React Native environment
-            window.ReactNativeWebView.postMessage('videoPlayed');
+            
+
```
Comment (kate-holdener) [inline lib/screens/AddNoteScreen.tsx:183]: Remove excessive blank lines

File: lib/screens/MorePage.tsx [hunk 6]
```diff
@@ -15,17 +15,21 @@ import { Ionicons } from "@expo/vector-icons";
...
 const { width, height } = Dimensions.get("window");
 
 export default function MorePage() {
 
   const { theme, isDarkmode, setIsDarkmode } = useTheme();
-  const {clearSession} = useAuth0();
 
   const toggleDarkmode = useTheme().toggleDarkmode;
+
+  //updated by karthik
```
Comment (kate-holdener) [inline lib/screens/MorePage.tsx:30]: This is not a good comment to have in the source code, we can use git log to figure out who updated which line (if we need to figure it out).

File: lib/screens/loginScreens/LoginScreen.tsx [hunk 7]
```diff
@@ -69,28 +69,39 @@ const LoginScreen: React.FC<LoginProps> = ({ navigation, route }) => {
   const onDismissSnackBar = () => toggleSnack(false);
 
   const handleLogin = async () => {
+
     if (username === "" || password === "") {
       toggleSnack(!snackState);
     } else {
+
+      // changes made by karthik
```
Comment (kate-holdener) [inline lib/screens/loginScreens/LoginScreen.tsx:77]: Please remove this, it clutters the source code.

**Comment by kate-holdener** [comment]
I'm confused why this PR shows so many changes and commits (and the commits from previous developers). What am I missing here?

File: lib/screens/MorePage.tsx (hunk 6 above)
Comment (rcAsironman) [reply lib/screens/MorePage.tsx:30]: Sure, Dr. Holdener, the final version of the code will not reflect the issues mentioned above.

**Comment by AmarHadzic** [comment]
> I'm confused why this PR shows so many changes and commits (and the commits from previous developers). What am I missing here?

@kate-holdener , I was tasked with fixing the login button while the keyboard was mounted. There were existing features that had a different approach than mine and I wanted to try this feature out.

**Comment by AmarHadzic** [comment]
hi @kate-holdener , your comments regarding comments in the code was a part of an old branch and not a part of this new PR. Would you be able to resolve those comments?

**Review by yashb196 — APPROVED** [review]
looks good

**Review by rcAsironman — APPROVED** [review]
lgtm;

**Pull request merged**